Dockerfile

```diff
@@ -3,14 +3,790 @@ ADD ./steamos /
+    sdl2 \
+    seatd \
+    serd \
+    shared-mime-info \
+    signon-kwallet-extension \
+    signon-plugin-oauth2 \
+    signon-ui \
+    signond \
+    slang \
+    smartmontools \
+    smbclient \
+    snappy \
+    socat \
+    solid \
+    sonnet \
+    sord \
+    sound-theme-freedesktop \
+    source-highlight \
+    spectacle \
+    speex \
+    speexdsp \
+    squashfs-tools \
+    sratom \
+    srt \
+    sshfs \
+    steam-im-modules \
+    steam-jupiter-stable \
+    steamdeck-kde-presets \
+    steamos-atomupd-client-git \
+    steamos-customizations-jupiter \
+    steamos-devkit-service \
+    steamos-efi \
+    strace \
+    svt-av1 \
+    syndication \
+    syntax-highlighting \
+    sysfsutils \
+    systemd-swap \
+    systemsettings \
+    taglib \
+    talloc \
+    tcl \
+    tdb \
+    tevent \
+    thin-provisioning-tools \
+    threadweaver \
+    tk \
+    trace-cmd \
+    tracker3 \
+    tree \
+    tslib \
+    ttf-dejavu \
+    ttf-hack \
+    ttf-twemoji-default \
+    udisks2 \
+    unrar \
+    unzip \
+    upower \
+    usbmuxd \
+    usbutils \
+    v4l-utils \
+    vid.stab \
+    vim \
+    vim-runtime \
+    vkmark-git \
+    vmaf \
+    volume_key \
+    vpower \
+    vulkan-icd-loader \
+    vulkan-radeon \
+    vulkan-tools \
+    wayland \
+    wayland-utils \
+    webrtc-audio-processing \
+    wget \
+    wireless_tools \
+    wireless-regdb \
+    wireplumber \
+    wpa_supplicant \
+    x264 \
+    x265 \
+    xbindkeys \
+    xbitmaps \
+    xcb-proto \
+    xcb-util \
+    xcb-util-cursor \
+    xcb-util-errors \
+    xcb-util-image \
+    xcb-util-keysyms \
+    xcb-util-renderutil \
+    xcb-util-wm \
+    xdg-dbus-proxy \
+    xdg-desktop-portal \
+    xdg-desktop-portal-kde \
+    xdg-user-dirs \
+    xdg-utils \
+    xdotool \
+    xf86-input-libinput \
+    xf86-video-amdgpu \
+    xfsprogs \
+    xkeyboard-config \
+    xorg-fonts-encodings \
+    xorg-server \
+    xorg-server-common \
+    xorg-setxkbmap \
+    xorg-xauth \
+    xorg-xdpyinfo \
+    xorg-xhost \
+    xorg-xkbcomp \
+    xorg-xmessage \
+    xorg-xprop \
+    xorg-xrandr \
+    xorg-xrdb \
+    xorg-xset \
+    xorg-xsetroot \
+    xorg-xwayland-jupiter \
+    xorg-xwininfo \
+    xorgproto \
+    xterm \
+    xvidcore \
+    xxhash \
+    zenity-light \
+    zeromq \
+    zimg \
+    zip \
+    zsh \
+    zxing-cpp \
  && sed -r -i 's/\[(jupiter|core|extra|community|multilib|holo)\]/\[\1-rel\]/g' /etc/pacman.conf \
  && pacman-key --init \
  && pacman-key --populate archlinux \
  && pacman-key --populate holo \
  && pacman -Sy \
 #  && comm -1 -2  <(pacman -Qeq | sort) <(pacman -Qoq /usr/include/ | sort) | pacman -S --noconfirm - \
- && comm -1 -2  <(pacman -Qdq | sort) <(pacman -Qoq /usr/include/ | sort) | pacman -S --noconfirm --asdeps - \
+ && comm -1 -2  <(pacman -Qdq | sort) <(pacman -Qoq /usr/include/ | sort) \
+    # Workaround for https://github.com/moby/buildkit/issues/1267
+    | grep -v '^filesystem$' \
```

Comment:
So the main problem in the linked issue is that Docker BuildKit mounts a custom `/etc/hosts` and `/etc/host.conf` (both readonly) into the container during the `docker build`'s `RUN` steps. When this `RUN` step tries to reinstall the `filesystem` package (which owns those two files), it fails because of the read-only mount from Docker.

So this workaround is just to skip reinstalling the `filesystem` package. It's probably not ideal, since some package files could technically be missing after the fact, but I don't think there's any way around the docker build limitation as linked. This is the verification output afterwards (which I think tracks with the contents of `.dockerignore`):
```console
# pacman -Qk filesystem
warning: filesystem: /boot/ (No such file or directory)
warning: filesystem: /boot/efi (No such file or directory)
warning: filesystem: /mnt (No such file or directory)
warning: filesystem: /usr/share/man/man2/ (No such file or directory)
warning: filesystem: /usr/share/man/man4/ (No such file or directory)
warning: filesystem: /usr/share/man/man6/ (No such file or directory)
filesystem: 125 total files, 6 missing files
```

Reply:
seems alright, thanks for explaining